source/common/http/http2/codec_impl.cc

```diff
@@ -130,6 +130,27 @@ void ConnectionImpl::StreamImpl::encodeTrailers(const HeaderMap& trailers) {
     parent_.sendPendingFrames();
   }
 }
+
+void ConnectionImpl::StreamImpl::encodeMetadata(const MetadataMap& metadata_map) {
+  if (!parent_.allow_metadata_) {
+    ENVOY_CONN_LOG(error, "Connection doesn't allow metadata", parent_.connection_);
```

Comment:
When do we envision this error happening? I'm mainly wondering if this should be a stat? Or should it just crash? Or just be a debug statement? I'm leery of error logs being useful to people.

Reply:
Correct my comments above: we don't need to check allow_metadata_ not because of the reasons I stated. If allow_metadata_ is false, nghttp2 will not pass metadata to codec. We don't need to check it in envoy code anywhere. Sorry about the wrong information!